### PR TITLE
feat(auth): add Clerk authentication to [POST] /api/v1/git_tokens/

### PR DESCRIPTION
**Pull request by mohamad-jaafar-monty**
Summary:
- updated API '[POST] /api/v1/git_tokens/' to start using Clerk authentication.

- Exception was being swallowed and not logged, added logging. 

- Fixed Unit Tests related to this API

- Moved the unit test to a proper structure

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved error handling and added structured logging for Git token endpoints to enhance error traceability.
- **Refactor**
	- Unified user authentication handling in Git token endpoints using authenticated user objects.
	- Optimized token masking logic and added a masked token field to store masked tokens securely.
- **New Features**
	- Introduced structured logging of exceptions across all Git token API endpoints.
- **Tests**
	- Added comprehensive unit tests for Git token endpoints, covering various success, failure, and edge cases.
	- Introduced fixtures for authenticated user mocking and encryption key patching.
	- Added extensive tests for authentication utilities and encryption helper functions.
	- Removed outdated and redundant test classes and fixtures related to token masking and authentication.
	- Reorganized and cleaned up test files for improved coverage and maintainability.
- **Chores**
	- Added a constant for encryption key error messages and updated encryption helper to use it.
	- Updated model to include masked token field for improved token security.

### CODE REVIEW

File: app/models/git_label.py [hunk 1]
```diff
@@ -26,6 +26,11 @@ class GitLabel(Model):
     token_value = fields.TextField(
         null=False, description="Access token for the git hosting service"
     )
+    
+    masked_token = fields.TextField(
+        null=False, description="Masked Access token for the git hosting service"
+    )
```
[the remark this answers is not in the document]
Comment (mohamad-jaafar-monty) [reply app/models/git_label.py]: done

File: app/routes/git_tokens.py [hunk 2]
```diff
@@ -86,9 +89,11 @@ async def handle_gitlab(
...
         )
-    except Exception:
+    except Exception as e:
+        logger.exception("Unexpected Failure while attempting to save GitLab token on Path = '[POST] /api/v1/git_tokens' -> handle_gitlab")
+        
         return APIResponse.error(
-            message=f"Failed to save GitLab token: {str(e)}",
+            message=constants.SERVICE_UNAVAILABLE,
             status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
         )
```
[the remark this answers is not in the document]
Comment (mohamad-jaafar-monty) [reply app/routes/git_tokens.py]: done

File: app/routes/git_tokens.py [hunk 3]
```diff
@@ -233,11 +241,7 @@ async def get_git_label_by_label(
                     "id": str(gl.id),
                     "label": gl.label,
                     "git_hosting": gl.git_hosting,
-                    "masked_token": mask_token(
-                        EncryptionHelper.decrypt(gl.token_value)
-                        if gl.token_value
-                        else ""
-                    ),
+                    "masked_token": gl.masked_token,
```
[the remark this answers is not in the document]
Comment (mohamad-jaafar-monty) [reply app/routes/git_tokens.py]: done
